rse-narrative.md

```diff
@@ -0,0 +1,157 @@
+book title, author and release date using the `head` command to select the relevant lines
+from an ebook text file. 
+(Some of the original text files will need to be edited very slightly
+to ensure that this information is on exactly the same line in each file.)
+Later on in the chapter,
+the content on `grep` could then show how it could have been used to locate the same
+information in the files 
+(i.e. so the information wouldn't need to be on exactly the same line in each file.)
+
+### Chapter 4: Command line programs
+
+This chapter currently involves a generic script called `script_template.py`,
+which can be configured using command line arguments or a configuration file.
+Instead of developing the generic `script_template.py`,
+this chapter could develop `countwords.py` instead.
+(The current version of `countwords.py` will need to be updated to use argparse
+and other libraries/concepts introduced in that chapter.)
+
+### Chapter 5: Git at the command line
+
+The narrative in this chapter currently involves Frances and her colleague Jean Jennings
+writing a history of the ENIAC project.
+Instead, in this chapter the `collate.py` script could be written from scratch,
+tracking changes with Git along the way.
+
+### Chapter 6: Advanced Git
+
+This chapter introduces the concept of branches.
+In the current draft, a violin plot is created on one branch
+and a beeswarm plot on the other.
+The better looking plot is then merged into the master branch.
+This example could be tweaked so that two slightly different versions of 
+`plotcounts.py` are developed instead
+(i.e. two different ways of plotting the word count data).
+
+### Chapter 7: Code style, review and refactor
+
+It's not clear how much of this chapter will ultimately end up in the novice Python book,
+but nonetheless code review will definitely be introduced here.
+The chapter text could be edited such that a draft version of `testfit.py`
+(with deliberate flaws written in)
+is presented and then reviewed/improved.
+
+### Chapter 8: Automating analyses 
+
+This chapter can stay pretty much as is.
+The introductory paragraphs about Zipf's Law can be removed 
+(because they should now appear in the introduction chapter)
+and the workflows developed in the chapter could be extended to also include
+`plotcounts.py` and `testfit.py`.
+
+### Chapter 10: Project structure
+
+Now that we've written all our scripts for the Zipf's Law analysis,
+this chapter should structure it all properly (i.e. with license files and README's etc).
+
+### Chapter 14: Correctness
+
+This chapter introduces testing and could be updated to write tests for the Zipf's Law code.
+For example, `countwords.py` could be tested using an input text file where the count
+for each word is already known.
+
+I think this chapter needs to come before the chapter on continuous integration,
+since it's these tests that Travis CI will implement.
+
+### Chapter 13: Python packaging and Chapter 12: R packaging
```

Comment:
I think that R packaging should be started from the beginning and start building from there (for the R split if we do it). We would have to make sure that we focus strongly on live-coding/exercises/practical side of R package development so we don't start mimicking the [R Packages](http://r-pkgs.had.co.nz/) book. With the inclusion of teams, collaboration, etc, I think we can differentiate ourselves enough.